components/newMentorProfile/sessions.js

```diff
@@ -0,0 +1,107 @@
+import React,{ useState } from 'react'
+import SessionCard from "../../components/mentorProfile/components/SessionCard";
+import styles1 from "../../styles/mentorTestimonial.module.css";
+function Sessions({mentorDetail}) {
+  const [modalPopup, setModalPopup] = useState(false);
+  const [selectedSession, setSelectedSession] = useState("");
+  const sendMail = async (data) => {
+    try {
+      setIsLoading(true);
+      const res = await axios.post(
+        `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/mentors/bookSessionMail`,
+        data
+      );
+      setIsLoading(false);
+      setModalPopup(false);
+      toast.success(
+        "Your session has been booked! Check your inbox for payment details."
+      ); // Success toast
+    } catch (error) {
+      setIsLoading(false);
+      if (error.response && error.response.status === 400) {
+        toast.error("You have already booked this session"); // Error toast
+      } else if (error.response && error.response.status === 405) {
+        toast.error("You are not allowed to book your own session"); // Error toast
+      } else {
+        console.error("Error sending mail:", error);
+        toast.error("Facing any problem? Email Us"); // Error toast
+      }
+    }
+  };
+  const handleClick = (mentordata) => {
+    const { sessionName, sessionMeetingDuration, priceSession } = mentordata;
+    const { email, name, username } = mentorDetail;
+
+    if (isUserLoggedIn) {
+      sessionStorage.removeItem("redirectUrl");
+      handleBookSession(
+        sessionName,
+        email,
+        name,
+        sessionMeetingDuration,
+        priceSession
+      );
+    } else {
+      const redirectUrl = window.location.href;
+      sessionStorage.setItem("redirectUrl", redirectUrl);
+      router.push(`/userAuth#login`);
+    }
+  };
+  const handleBookSession = async (
```

Comment:
this whole functionality is part of [mentorusername].page.js...I wonder if it is working?

Reply:
yes it is. I'll remove it from sessions.js